UniswapV2Adapter/contracts/UniswapV2Adapter.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./interfaces/IUniswapFactory.sol";
+import "./interfaces/IUniswapPair.sol";
+import "./interfaces/IERC20.sol";
+import "./lib/SafeERC20.sol";
+import "./BaseAdapter.sol";
+
+contract UniswapV2Adapter is BaseAdapter {
+    using SafeERC20 for IERC20;
+
+    uint256 internal constant FEE_DENOMINATOR = 1e3;
+    uint256 public immutable feeCompliment;
+    address public immutable factory;
+
+    constructor(
+        string memory _name,
+        address _factory,
+        uint256 _fee,
+        uint256 _swapGasEstimate
+    ) BaseAdapter(_name, _swapGasEstimate) {
+        feeCompliment = FEE_DENOMINATOR - _fee;
+        factory = _factory;
```

Comment:
Severity: Low
Confidence: Medium

```
UniswapV2Adapter.constructor(string,address,uint256,uint256)._factory lacks a zero-check on :
                - factory = _factory
```
                
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#missing-zero-address-validation

Reply:
Severity: Medium
Confidence: Medium

```
UniswapV2Adapter.constructor(string,address,uint256,uint256)._factory lacks a fee amount check on :
                - feeCompliment = FEE_DENOMINATOR - _fee;
```

Recommendation: Add a condition check: _fee must be equal and smaller than 1e3